### PR TITLE
Perf: Enable AoT compilation 

### PR DESCRIPTION
**Pull request by robwormald**
Initial pass at enabling AoT compilation and lazy-loading of code.

Changes required:
- adds a new `webpack.aot.js` config file and `npm run build:aot` script
- remove `require('./some-template.html')` as its not AoT compatible (will need to add loader for dev mode)
- moved routes/policy-guide and /explore-code to standalone modules 
- removed unused dependencies

Brings initial bundle down to ~100k gzip with significantly less work on the main thread due to compilation happening ahead of time.

### CODE REVIEW

File: config/webpack.prod.js [hunk 1]
```diff
@@ -216,17 +217,6 @@ module.exports = function (env) {
...
         src: 'index.html'
       }),
-
-      new CopyWebpackPlugin([
-       {
-         from: 'src/assets',
-         to: 'assets',
-       },
-       {
-         from: 'config/CNAME'
```
Comment (lukad03) [inline config/webpack.prod.js]: We need the CNAME file to be copied over for Production purposes. Do you plan on adding this functionality back in?

**Comment by lukad03** [comment]
I've pulled the branch to check out what's happening locally - seems the test failures are consistent with what's happening in CI.

**Comment by lukad03** [comment]
Seems like this is the same issue as the Angular2 Webpack Starter: https://github.com/AngularClass/angular2-webpack-starter/issues/1021

**Comment by cdoremus** [comment]
What is the status of this work? 

I believe that the project would benefit from AOT, but perhaps this change could be eased-in incrementally. 

Here are a few of suggestions:
* Change the @Component attributes styles to styleUrls and template to templateUrl using the 
webpack loaders angular2-template-loader and raw-loader. I think this would have to be done to all components at once.
* Break code into separate @NgModule modules as a prerequisite to implementing Angular Router lazy loading. This would require some discussion as to the best module structure, but I believe that each module can be created separately prior to implementing lazy loading.
* Go through the AOT checklist here (https://github.com/rangle/angular-2-aot-sandbox) to make sure the code in this repo does not use AOT-incompatible practices.

**Comment by robwormald** [comment]
Closing this as its stale. Will re-open with a more incremental

**Pull request closed without merging**